hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -210,6 +210,13 @@ public Response put(
             "Connection", "close").build();
       }
 
+      // Normal put object
+      bucket = getBucket(bucketName);
+      Map<String, String> customMetadata =
+          getCustomMetadataFromHeaders(headers.getRequestHeaders());
```

Comment:
Filter the metadata here instead of filtering it in Key creation. Filtering it in Key construction might have unintended consequences if the key is supposed to have one of the reserved metadata keys.

Reply:
Make sense! Sure!!